jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -50,7 +71,14 @@ def parse_data(app, file_obj, ext=None, data_label=None):
                 pf = rgb2gray(im)
             pf = pf[::-1, :]  # Flip it
             _parse_image(app, pf, data_label, ext=ext)
-        else:  # Assume FITS
+        elif file_obj_lower.endswith('.asdf'):
+            if rdd is not None:
```

Comment:
What if someone passes in ASDF file that isn't Roman? Is there a way to identify that it is actually Roman data before attempting to read with Roman package?

Reply:
The [`roman_datamodels` docs page on the structure of Roman data products](https://roman-datamodels.readthedocs.io/en/latest/roman_datamodels/datamodels/general_structure.html#overview) suggests that it is a "convention" that data are stored under a `roman` attribute. I've confirmed this in the L2 products. If you run:
```python
import asdf

with asdf.open(path) as f:
    f.info()
```
you'll see
```
root (AsdfObject)
├─asdf_library (Software)
│ ├─author (str): The ASDF Developers
│ ├─homepage (str): http://github.com/asdf-format/asdf
│ ├─name (str): asdf
│ └─version (str): 2.13.0
├─history (dict)
│ └─extensions (list) ...
└─roman (WfiImage) # The schema for WFI Level 2 images.
...
```
Maybe the right thing to do is (1) open the file with the `asdf` package, (2) check for a `roman` attribute and load with `roman_datamodels` if so, (3) otherwise try passing the ASDF file without `roman_datamodels`.